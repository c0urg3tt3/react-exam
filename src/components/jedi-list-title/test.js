import React from 'react'
import { shallow } from 'enzyme'

import JediListTitle from './component'

describe('[component] JediListTitle', () => {
  it('should render component', () => {
    const wrapper = shallow(<JediListTitle/>)

    expect(wrapper.exists()).toEqual(true)
    expect(wrapper.is('li')).toEqual(true)
    expect(wrapper.hasClass('jedi-list-title')).toEqual(true)
    expect(wrapper.children()).toHaveLength(2)

    const id = wrapper.find('.jedi-list-title_id')
    expect(id.is('span')).toBe(true)
    expect(id.text()).toBe("id")

    const name = wrapper.find('.jedi-list-title_name')
    expect(name.is('span')).toBe(true)
    expect(name.text()).toBe("name")

    expect(wrapper.html()).toMatchSnapshot()
  })
})
