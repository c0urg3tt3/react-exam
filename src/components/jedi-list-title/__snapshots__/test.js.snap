// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[component] JediListTitle should render component 1`] = `"<li class=\\"jedi-list-title\\"><span class=\\"jedi-list-title_id\\">id</span><span class=\\"jedi-list-title_name\\">name</span></li>"`;
